File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.java
class Solution {
    public String firstPalindrome(String[] words) {
        for(int i=0; i<words.length; i++){
            String str=words[i];
            int s=0, e=str.length()-1;
            while(s<e){
                if(str.charAt(s)!=str.charAt(e))break;
                s++;
                e--;
            }
            if(s>=e)return str;
        }
        return "";
    }
}

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for i in words:
            s,e=0,len(i)-1
            while(s<e):
                if i[s]!=i[e]:
                    break
                s+=1
                e-=1
            if s>=e:
                return i
        return ""